test/src/bgw/scheduler_mock.c

```diff
@@ -196,7 +196,9 @@ static bool
 test_job_3_long()
 {
 	BackgroundWorkerBlockSignals();
-	prev_signal_func = pqsignal(SIGTERM, log_terminate_signal);
+	/* Only set prev_signal_func once to prevent it from being set to log_terminate_signal). */
```

Comment:
Not sure I understand this comment, since we seem to be setting it to `log_terminate_signal` ourselves? Also stray `)` at the end.

Reply:
It *should* get set to the previous `SIGTERM` handler. It should never get set to `log_terminate_signal`, otherwise it will call itself recursively forever.
Fixed stray `)`.